lib/ddtrace/sampling/token_bucket.rb

```diff
@@ -98,5 +98,21 @@ def increment_conforming_count
         @conforming_messages += 1
       end
     end
+
+    # TODO: This class name is so bad, yet so good.
+    # [Class documentation]
+    class UnlimitedLimiter < RateLimiter
```

Comment:
Haha, great name. If you want a different name my suggestions are just as quirky: `NonLimiter`, `NoLimiter`, `NeverLimiter`. But honestly, `UnlimitedLimiter` works just fine.

I might suggest though that this behavior for a limiter shouldn't be a `Limiter` but a composable module for a `Limiter` (e.g. `RateLimiiter.new.extend(UnlimitedRate)`) because an unlimited limiter isn't a distinct species of limiter, just one that behaves a particular way.

Reply:
I do see the point you are making, specially if we had something like a `DebugRateLimiter` that prints all operations to stdout: we would be able to add this functionality to an existing rate limiter as a mixin.

But for the case of `UnlimitedLimiter`, I think that being subclass of `RateLimiter` makes more sense, as it pretty much overrides all behaviour of `RateLimiter` and replaces with its own. I see it less as a composable feature-set and more as a whole type of limiter in itself. When overlaying it with the token bucket implementation, for example, it removes all token bucket functionality and replaces with its own implementation.

Reply:
Alright, its probably fine using subclasses here, too. Just wanted to consider our options, but I'm cool with this.